packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -212,30 +251,40 @@ class RenderParagraph extends RenderBox {
     markNeedsLayout();
   }
 
-  void _layoutText({ double minWidth = 0.0, double maxWidth = double.infinity }) {
-    final bool widthMatters = softWrap || overflow == TextOverflow.ellipsis;
-    _textPainter.layout(minWidth: minWidth, maxWidth: widthMatters ? maxWidth : double.infinity);
-  }
-
-  void _layoutTextWithConstraints(BoxConstraints constraints) {
-    _layoutText(minWidth: constraints.minWidth, maxWidth: constraints.maxWidth);
-  }
-
   @override
   double computeMinIntrinsicWidth(double height) {
-    _layoutText();
-    return _textPainter.minIntrinsicWidth;
+    if (!_canComputeIntrinsics())
+      return 0;
+    if (_needsLayout) {
+      _computeChildrenWidthWithMinIntrinsics(height);
+      _layoutText();
+      _needsLayout = true; // Purposefully not markNeedsLayout(). Does not call super.
```

Comment:
that comment isn't useful as written. It says what it does, but we can see what it does. _Why_ does it do it?

Reply:
(same below)